python/astro_metadata_translator/bin/translate.py

```diff
@@ -144,12 +144,19 @@ def read_file(
         if output_mode == "auto":
             output_mode = "table" if len(headers) > 1 else "verbose"
 
+        def _filter_none(info: ObservationInfo, attr: str) -> Any:
+            val = getattr(info, attr)
+            if val is None:
+                return "-"
```

Comment:
Will that work if the expected column type is numeric and tries to format it with `%f` specification?

Reply:
Oops. You are right. I need to do the formatting inside this. All my tests were with string types.

Reply:
I am thinking I should be using a masked `astropy.table` here...

Reply:
@andy-slac can you take another look. To use QTable I had to reorganize how files are encountered. It's a bit of a hack but the output is nicer:

```
       ObsId        ImgType               Object                           Filter              Detector  ExpTime 
                                                                                                            s    
------------------- ------- --------------------------------- -------------------------------- -------- ---------
            1038843      --                           w2.+2+2                         i.MP9702    ccd02       nan
ct4m20121211t220632    zero                    postflats-BIAS              solid plate 0.0 0.0       S1       0.0
ct4m20150220t004721 science                       Blind15A_03                                g      S29      86.0
ct4m20190402t050618 science                                       VR DECam c0007 6300.0 2600.0      S29     120.0
ct4m20130213t090140 science                 Trifid and Lagoon g DECam SDSS c0001 4720.0 1520.0       S1      30.0
ct4m20130901t060255 science DES supernova hex SN-S1 tiling 22 z DECam SDSS c0004 9260.0 1520.0       S1     200.0
       HSCA04090000 science                          SSP-Wide                            HSC-R     1_07     150.0
       HSCA90402400 science                         STRIPE82L                            HSC-I     1_12      30.0
             849375 science                                D3                                r    ccd99   300.202
            1038843 science                           w2.+2+2                         i.MP9702    ccd02   615.037
       6377 4 g 407 science                              82 S                                g       g4 53.907456
       SUPE00535770 science               Ecliptic Deep Field                           W-S-R+ nausicaa     200.0
```

Reply:
Right-aligned strings in formatted tables hurt my eyes 🙂 Also in `butler` output.

Reply:
Here you go:

```
       ObsId        ImgType               Object                           Filter              Detector  ExpTime 
                                                                                                            s    
------------------- ------- --------------------------------- -------------------------------- -------- ---------
1038843             --      w2.+2+2                           i.MP9702                            ccd02       ———
ct4m20121211t220632 zero    postflats-BIAS                    solid plate 0.0 0.0                    S1       0.0
ct4m20150220t004721 science Blind15A_03                       g                                     S29      86.0
ct4m20190402t050618 science                                   VR DECam c0007 6300.0 2600.0          S29     120.0
ct4m20130213t090140 science Trifid and Lagoon                 g DECam SDSS c0001 4720.0 1520.0       S1      30.0
ct4m20130901t060255 science DES supernova hex SN-S1 tiling 22 z DECam SDSS c0004 9260.0 1520.0       S1     200.0
HSCA04090000        science SSP-Wide                          HSC-R                                1_07     150.0
HSCA90402400        science STRIPE82L                         HSC-I                                1_12      30.0
849375              science D3                                r                                   ccd99   300.202
1038843             science w2.+2+2                           i.MP9702                            ccd02   615.037
6377 4 g 407        science 82 S                              g                                      g4 53.907456
SUPE00535770        science Ecliptic Deep Field               W-S-R+                           nausicaa     200.0
```